libsolidity/parsing/Parser.cpp

```diff
@@ -257,57 +257,67 @@ ASTPointer<ContractDefinition> Parser::parseContractDefinition()
 {
 	RecursionGuard recursionGuard(*this);
 	ASTNodeFactory nodeFactory(*this);
+	ASTPointer<ASTString> name;
```

Comment:
Variables should be declared as close to their use as possible.

Reply:
I am confused here. Where exactly would you have this moved to? 

If I understand C++ right, the declaration has to be outside of the try block because that is where it is used. It is initialized inside the try block. 

However it is true that these probably have to be initialized to some value in case there is an unexpected fatalError inside the try.

Reply:
Oh I'm sorry, I didn't see that it is actually used further down.